frame/dvm/ethereum/src/lib.rs

```diff
@@ -302,13 +302,19 @@ pub mod pallet {
 	}
 
 	#[pallet::event]
-	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	#[pallet::generate_deposit(pub fn deposit_event)]
 	/// Ethereum pallet events.
-	pub enum Event {
+	pub enum Event<T: Config> {
 		/// An ethereum transaction was successfully executed. \[from, to/contract_address, transaction_hash, exit_reason\]
 		Executed(H160, H160, H256, ExitReason),
 		/// DVM Transfer succeeded. \[from, to, value\]
 		DVMTransfer(H160, H160, U256),
+		/// Withdraw RING from DVM to darwinia account. \[sender, destination, value\]
+		RingBack(H160, T::AccountId, U256),
+		/// Transfer substrate account KTON to WKTON contract. \[sender, value\]
+		TransferToWKton(H160, U256),
+		/// Withdraw from WKON contract to substrate account. \[withdraw, value\]
+		WithdrawFromWKton(T::AccountId, U256),
```

Comment:
Are these events added on the request of Subscan? I'm not sure whether they are satisfy the requirements. @freehere107

Reply:
This request comes from the smart app @sxlwar. I have discussed this with him, it works.

Reply:
It is better to have the event of KTON pallet transfer here

Reply:
Native KTON transfer will trigger events from the balances pallet. In the current precompile scenario, this transfer occurs between the substrate native side and the WKTON contract. I do not get the meaning of the event you said above.